src/decision.rs

```diff
@@ -1,85 +1,22 @@
-use std::collections::{BTreeMap, BTreeSet};
+use std::collections::BTreeMap;
 
 use blsttc::{PublicKeySet, Signature};
-use log::warn;
 use serde::{Deserialize, Serialize};
 
-use crate::{Error, Fault, Generation, NodeId, Proposition, Result, SignedVote, VoteCount};
+use crate::{verify_sig, Generation, Proposition, Result};
 
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 pub struct Decision<T: Proposition> {
-    pub votes: BTreeSet<SignedVote<T>>,
+    pub generation: Generation,
     pub proposals: BTreeMap<T, Signature>,
-    pub faults: BTreeSet<Fault<T>>,
 }
 
 impl<T: Proposition> Decision<T> {
     pub fn validate(&self, voters: &PublicKeySet) -> Result<()> {
-        let all_votes = self.votes_by_voter();
-        let known_faulty_voters = self.faulty_ids();
-        let expected_generation = self.generation()?;
-
-        for vote in self.votes.iter() {
-            vote.validate(voters, &Default::default())?;
-
-            if vote.vote.gen != expected_generation {
-                warn!("Not all votes in decision are from same generation");
-                return Err(Error::InvalidDecision);
-            }
-
-            if let Err(faults) =
-                vote.detect_byzantine_faults(voters, &all_votes, &Default::default())
-            {
-                let detected_faults = BTreeSet::from_iter(faults.into_keys());
-                if !detected_faults.is_subset(&known_faulty_voters) {
-                    warn!("Detected faulty voters who is not present in faults");
-                    return Err(Error::InvalidDecision);
-                }
-            }
-        }
-
-        for fault in self.faults.iter() {
-            fault.validate(voters).map_err(Error::FaultIsFaulty)?;
-        }
-
-        let vote_count = VoteCount::count(self.votes.iter().cloned(), &self.faulty_ids());
-        if let Some(proposals) = vote_count.get_decision(voters)? {
-            if proposals != self.proposals {
-                warn!("Proposals from votes does not match decision proposals");
-                return Err(Error::InvalidDecision);
-            }
-        } else {
-            warn!("Votes in decision don't form a decision");
-            return Err(Error::InvalidDecision);
+        for (proposal, sig) in self.proposals.iter() {
+            verify_sig(proposal, sig, &voters.public_key())?;
```

Comment:
actually now we only need to pass in the section public key right?
maybe better to change it as well in this PR ?
as it could happen that the caller only has the knowledge of section_public_key, instead of the full key set.